dataplane/src/drivers/kernel.rs

```diff
@@ -130,100 +137,217 @@ fn get_interface_ifindex(interfaces: &[Interface], name: &str) -> Option<u32> {
+        workers: usize,
         setup_pipeline: Arc<dyn Send + Sync + Fn() -> DynPipeline<TestBuffer>>,
     ) {
-        let mut pipeline = setup_pipeline();
-
-        // Spawn threads in this function, e.g.,
-        // let thandle = concurrency::thread::spawn(move || {
-        //     let mut pipeline = setup_pipeline();
-        //     ...
-        // });
+        // Prepare interfaces/poller
+        let mut kiftable = match build_kif_table(args) {
+            Ok(t) => t,
+            Err(e) => {
+                error!("Failed to initialize kernel interface table: {e}");
+                return;
+            }
+        };
 
-        /* build kernel interface table from interfaces available and cmd line args */
-        let mut kiftable = build_kif_table(args);
+        // Spawn workers
+        let (to_workers, from_workers) = match Self::spawn_workers(workers, setup_pipeline) {
+            Ok(chans) => chans,
+            Err(e) => {
+                error!("Failed to start workers: {e}");
+                return;
+            }
+        };
+        let buckets = to_workers.len().max(1);
+        let poll_timeout = Some(Duration::from_millis(2));
 
-        /* poll the registered interfaces */
-        let mut events = Events::with_capacity(64);
+        // Dispatcher loop: drain processed packets, poll RX, parse+shard, TX results.
+        let mut events = Events::with_capacity(256);
         loop {
-            if let Err(e) = kiftable.poll.poll(&mut events, None) {
+            // 1) Drain processed packets coming back from workers, serialize + TX
+            while let Ok(mut pkt) = from_workers.try_recv() {
+                // choose outgoing interface from meta
+                let oif_id_opt = pkt.get_meta().oif.as_ref().map(InterfaceId::get_id);
+                if let Some(oif_id) = oif_id_opt {
+                    if let Some(outgoing) = kiftable.get_mut_by_index(oif_id) {
+                        match pkt.serialize() {
+                            Ok(out) => {
+                                if let Err(e) = outgoing.sock.write_all(out.as_ref()) {
+                                    error!("TX failed on '{}': {e}", &outgoing.name);
+                                } else {
+                                    debug!(
+                                        "TX {} bytes on interface {}",
+                                        out.as_ref().len(),
+                                        &outgoing.name
+                                    );
+                                }
+                            }
+                            Err(e) => error!("Serialize failed: {e:?}"),
+                        }
+                    } else {
+                        warn!("TX drop: unknown oif {}", oif_id);
+                    }
+                } else {
+                    // No oif set; nothing to send (likely dropped upstream).
+                    debug!("No oif in packet meta; dropping");
```

Comment:
Is this an error?  Doesn't the packet have a drop flag we can check to know if it is supposed to be dropped?